admin/services/pupil.service.js

```diff
@@ -38,4 +38,34 @@ pupilService.getPupilsWithFullNames = async (schoolId) => {
   }))
 }
 
+/**
+ * Fetch all pupils for a school by schoolId sorted by lastname ascending.
+ * Sorting to be removed in a future version
+ * @param schoolId required
+ * @returns {Promise<*>}
+ */
+pupilService.findPupilsBySchoolId = function findPupilsBySchoolId (schoolId) {
+  if (!schoolId) {
+    throw new Error('schoolId is required')
+  }
+  return pupilDataService.sqlFindPupilsBySchoolId(schoolId)
```

Comment:
There is an opportunity here to remove the ordering from the database and move it to this service.  Interested?

Reply:
oh yes, back into dev we go....